assets/app/views/create.html

```diff
@@ -10,156 +10,199 @@
+                          </catalog-image>
+
+                          <!-- Add an empty item to keep the last tile from stretching to 100% width. -->
+                          <div style="height: 0;" class="tile-image"></div>
+                        </div>
                       </div>
                     </div>
-                  </div>
-
-                  <!-- Don't show images without the builder tag by default, by allow the user to pick one. -->
-                  <div ng-if="filteredNonBuilders.length" click-to-reveal link-text="Don't see the image you are looking for?" class="gutter-bottom">
-                    <h2>Additional Images</h2>
-                    <div class="gutter-bottom">
-                      <span class="pficon pficon-warning-triangle-o"></span>
-                      Some images in this list may not be able to build source. Use with caution.
-                    </div>
-                    <div class="catalog catalog-fluid">
-                      <catalog-image
-                          image-stream="image.imageStream"
-                          image-tag="image.imageStreamTag"
-                          project="{{projectName}}"
-                          version="image.version"
-                          ng-repeat="image in filteredNonBuilders | orderBy : ['name', 'imageStream.metadata.namespace']">
-                      </catalog-image>
-
-                      <!-- Add an empty item to keep the last tile from stretching to 100% width. -->
-                      <div style="height: 0;" class="tile-image"></div>
+                  </uib-tab>
+
+                  <uib-tab active="selectedTab.fromFile">
+                    <uib-tab-heading>From File</uib-tab-heading>
+                    <div class="row">
+                      <div class="col-sm-12 from-file">
+                        <form name="form">
+                          <osc-file-input model="editor.content" drop-area=".from-file" help-text="Upload file by dragging & dropping,  selecting it, or pasting from the clipboard."></osc-file-input>
+                          <div ui-ace="{
+                            mode: 'yaml',
+                            theme: 'eclipse',
+                            onLoad: aceLoaded,
+                            onChange: aceChanged,
+                            rendererOptions: {
+                              fadeFoldWidgets: true,
+                              showPrintMargin: false
+                            }
+                          }" ng-model="editor.content" class="editor ace_editor-bordered yaml-mode" id="add_component_editor" required></div>
+                          <div class="buttons gutter-top-bottom">
+                            <button
+                              type="submit"
+                              ng-click="create()"
+                              ng-disabled="editor.empty"
+                              class="btn btn-primary btn-lg">
+                              Add
```

Comment:
Should probably be "Create" to match `oc create`. Also per Patternfly guidelines:

https://www.patternfly.org/styles/terminology-and-wording/#action-labels